### ISSUE TITLE
pubsub emulator: disable auth

### ISSUE THREAD

**Issue by broady** [GoogleCloudPlatform/google-cloud-go#280]
DefaultClient always tries to hit the internet.

When we're in emulator mode, we should do something like:
```
cloud.WithTokenSource(oauth2.StaticTokenSource(&oauth2.Token{}))
```

**Comment by okdave** [comment]
Most emulators don't need any auth, so we should be able to just skip the oauth layer altogether?

**Comment by broady** [comment]
Yep, just using http.DefaultClient is even better.

**Comment by broady** [comment]
datastore already bypasses, sending a change for pub/sub.

**Comment by broady** [comment]
Fixed in 423496d4595577faf48edc140ed0ed3a63119d49
